File: 백준/Silver/3986. 좋은 단어/좋은 단어.cc
#include <bits/stdc++.h>
using namespace std;

int N, ret;
string temp;
int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    cin >> N;

    for (int i = 0; i < N; i++)
    {
        stack<char> st;
        cin >> temp;
        st.push(temp[0]);
        for (int j = 1; j < temp.length(); j++)
        {
            if(!st.empty() && temp[j] == st.top()){
                st.pop();
            }else{
                st.push(temp[j]);
            }
        }

        if(st.size() == 0){
            ret ++;
        }
    }
    cout << ret;

    return 0;
}

File: 백준/Silver/3986. 좋은 단어/좋은 단어.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        // StringTokenizer st = new StringTokenizer(br.readLine());
        // int N = Integer.parseInt(st.nextToken());
        int N = Integer.parseInt(br.readLine());
        int ret = 0;
        for (int i = 0; i < N; i++){
            String s = br.readLine();
            Stack<Character> st = new Stack<>();
            for (int j = 0; j < s.length(); j++){
                if (st.empty() || st.peek() != s.charAt(j)){
                    st.push(s.charAt(j));
                }else{
                    st.pop();
                }
            }
            if (st.empty()){
                ret++;
            }
        }
        
        System.out.println(ret);


    }
}